docs/pages/guides/server-components.mdx

```diff
@@ -0,0 +1,395 @@
+- Data fetching with async components and React Suspense.
+- Working with secrets and server-side APIs.
+- Server-side rendering (SSR) for SEO and performance.
+- Build-time rendering to remove unused JS code.
+
+Expo Router enables support for [React Server Components](https://react.dev/reference/rsc/server-components) on all platforms. This is an early preview of a feature that will be enabled by default in Expo Router.
+
+## Usage
+
+Your project must use Expo Router and React Native new architecture. New Architecture is not supported in Expo Go, so you'll need to [create a development build](/develop/development-builds/introduction/) or use the web.
+
+As of SDK 52, there is no strategy for migrating existing projects. We recommend starting with a new project.
+
+Enable the flag in the project app config:
+
+```json app.json
+{
+  "expo": {
+    "experiments": {
+      "reactServerComponents": true
+    }
+  }
+}
+```
+
+Ensure the entry module is `expo-router/entry` (default) in **package.json**. Then, create an initial route **app/index.tsx**:
+
+```tsx app/index.tsx
+// This is a server component.
+import { Text } from 'react-native';
+
+export default function Index() {
+  return <Text>Hello, world!</Text>;
+}
+```
+
+## Navigating between routes
+
+You can navigate between routes by using the `<Link />` API from Expo Router (not all props are supported yet).
+
+```tsx app/index.tsx
+import { Link } from 'expo-router';
+
+export default function Index() {
+  return (
+    <Link href="/second">
+      <Text>Hello, world!</Text>
+    </Link>
+  );
+}
+```
+
+## Layout Routes
+
+> Complex built-in layouts such as Stack and Tabs are not yet supported. They will fallback to a `<Slot />`.
+
+Layout routes (**app/_layout.tsx**) can wrap all routes in a common layout.
+
+```tsx app/_layout.tsx
+import { View } from 'react-native';
+import { Slot } from 'expo-router';
+
+export default function Layout() {
+  return (
+    <View>
+      <Slot />
+    </View>
+  );
+}
+```
+
+The `Slot` component is used to render the current route.
+
+## Server Components
+
+Server Components run in the server, meaning they can access Node.js built-ins and server APIs. They can also use async components.
+
+Consider the following component which fetches data and renders it:
+
+```tsx app/index.tsx
+import { Image, Text, View } from 'react-native';
+
+export async function Pokemon() {
+  const res = await fetch('https://pokeapi.co/api/v2/pokemon/2');
+  const json = await res.json();
+  
+  return (
+    <View style={{ padding: 8, borderWidth: 1 }}>
+      <Text style={{ fontWeight: 'bold', fontSize: 24 }}>{json.name}</Text>
+      <Image source={{ uri: json.sprites.front_default }} style={{ width: 100, height: 100 }} />
+
+      {json.abilities.map(ability => (
+        <Text key={ability.ability.name}>- {ability.ability.name}</Text>
+      ))}
+    </View>
+  );
+}
+```
+
+- You cannot use hooks like `useState`, `useEffect`, or `useContext` in Server Components.
```

Comment:
```suggestion
### Key points

- You cannot use hooks like `useState`, `useEffect`, or `useContext` in Server Components.
```

Let's add a sub-section so that people don't skim over this.